src/main/java/uk/gov/justice/hmpps/prison/repository/converter/YesNoToBooleanConverter.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.justice.hmpps.prison.repository.converter;
+
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+
+@Converter
+public class YesNoToBooleanConverter implements AttributeConverter<Boolean, String> {
+    @Override
+    public String convertToDatabaseColumn(Boolean attribute) {
+        return attribute ? "Y" : "N";
+    }
+
+    @Override
+    public Boolean convertToEntityAttribute(String dbData) {
+        return dbData.equalsIgnoreCase("Y");
```

Comment:
bit confused about this - won't this blow up in a heap if `dbData` is `null`?  Also why return an object `Boolean` rather than the primitive - this suggests that it could be `null`?

Reply:
Thanks for the catch on the null edge case. 

For the object vs primitive Boolean. This class implements the `AttributeConverter`. I expects 2 generic params, 1 is the  converted type and the other is for the database column type: `AttributeConverter<Boolean, String>`.

`convertToEntityAttribute` is the concrete implementation off the interface and it expects the return type to match the provided generic type.